distributor/distributor.go

```diff
@@ -452,14 +453,18 @@ func (d *Distributor) forAllIngesters(f func(cortex.IngesterClient) (interface{}
 	}
 
 	var lastErr error
-	result := []interface{}{}
+	result, numErrs := []interface{}{}, 0
 	for range ingesters {
 		select {
 		case resp := <-resps:
 			result = append(result, resp)
 		case lastErr = <-errs:
+			numErrs++
 		}
 	}
+	if numErrs > (d.cfg.ReplicationFactor - d.cfg.MinReadSuccesses) {
```

Comment:
This function is about reading from all the ingesters (to fetch data such as all label names).  If we only read from MinReadSuccesses (ie 2), then we wouldn't touch anyway near all ingesters.  However, due to replication, it is safe to ignore errors from `floor(RF/2)`, which in this case is the same as `ReplicationFactor - d.cfg.MinReadSuccesses` - mainly because its guaranteed by the way we write that every piece of data lives on at least `ceil(RF/2)` ingesters.